runtime/src/snapshot_utils.rs

```diff
@@ -809,25 +1243,133 @@ fn rebuild_bank_from_snapshots(
+    debug!(
+        "bprumo DEBUG rebuild_bank_from_snapshots()\n\tfss snapshot_paths: {:?}\n\tiss snapshot paths: {:?}",
+        fss_snapshot_paths,
+        iss_snapshot_paths
+    );
+    if fss_snapshot_paths.len() > 1 || iss_snapshot_paths.len() > 1 {
         return Err(get_io_error("invalid snapshot format"));
     }
-    let root_paths = snapshot_paths
+
+    let fss_root_paths = fss_snapshot_paths
+        .pop()
+        .ok_or_else(|| get_io_error("No snapshots found in snapshots directory"))?;
+
+    let iss_root_paths = iss_snapshot_paths
         .pop()
         .ok_or_else(|| get_io_error("No snapshots found in snapshots directory"))?;
 
     info!(
-        "Loading bank from {}",
-        &root_paths.snapshot_file_path.display()
+        "Loading bank from {} and {}",
+        &fss_root_paths.snapshot_file_path.display(),
+        &iss_root_paths.snapshot_file_path.display()
     );
-    let bank = deserialize_snapshot_data_file(&root_paths.snapshot_file_path, |mut stream| {
-        Ok(match snapshot_version_enum {
-            SnapshotVersion::V1_2_0 => bank_from_stream(
-                SerdeStyle::Newer,
-                &mut stream,
-                account_paths,
-                unpacked_append_vec_map,
+
+    debug!(
+        "bprumo DEBUG: rebuild_bank_from_incremental_snapshots(), Loading bank from {} and {}",
+        &fss_root_paths.snapshot_file_path.display(),
+        &iss_root_paths.snapshot_file_path.display()
+    );
+
+    let bank = deserialize_incremental_snapshot_data_file(
+        &fss_root_paths.snapshot_file_path,
+        &iss_root_paths.snapshot_file_path,
+        |mut fss_stream, mut iss_stream| {
+            Ok(match snapshot_version_enum {
+                SnapshotVersion::V1_2_0 => bank_from_stream_incremental(
+                    SerdeStyle::Newer,
+                    &mut fss_stream,
+                    &mut iss_stream,
+                    account_paths,
+                    unpacked_append_vec_map,
+                    genesis_config,
+                    frozen_account_pubkeys,
+                    debug_keys,
+                    additional_builtins,
+                    account_indexes,
+                    accounts_db_caching_enabled,
+                    limit_load_slot_count_from_snapshot,
+                    shrink_ratio,
+                ),
+            }?)
+        },
+    )?;
+
+    let status_cache_path = iss_unpacked_snapshots_dir.join(SNAPSHOT_STATUS_CACHE_FILE_NAME);
+    let slot_deltas = deserialize_snapshot_data_file(&status_cache_path, |stream| {
+        info!(
+            "Rebuilding status cache from {}",
+            status_cache_path.display()
+        );
+        let slot_deltas: Vec<BankSlotDelta> = bincode::options()
+            .with_limit(MAX_SNAPSHOT_DATA_FILE_SIZE)
+            .with_fixint_encoding()
+            .allow_trailing_bytes()
+            .deserialize_from(stream)?;
+        Ok(slot_deltas)
```

Comment:
maybe a comment here about why we take the status cache from the incremental snapshot and not the full snapshot

Reply:
Good point, will do!